staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -198,8 +199,6 @@ func (c *CacheDelegator) GetList(ctx context.Context, key string, opts storage.L
 			return c.storage.GetList(ctx, key, opts, listObj)
 		}
 	}
-	requestWatchProgressSupported := etcdfeature.DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress)
-	consistentRead := opts.ResourceVersion == "" && utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) && requestWatchProgressSupported
```

Comment:
Issue is with this logic. `opts.ResourceVersion == ""` is not enough to distinguish consistent read, we technically also need to validate `optsResourceVersionMatch == "" && opts.Predicate.Continue == ""` to be really sure this is consistent read.